core/app/models/spree/stock/splitter/weight.rb

```diff
@@ -25,20 +19,20 @@ def reduce(package)
           # This also prevents an additional package if no fit is possible
           package.contents.clear
           package.contents << contents.shift
-          _split_packages = [package]
-          while contents.present?
+          split_packages = [package]
 
-            package_to_use = choose_package _split_packages, contents.first
+          while contents.present?
+            package_to_use = choose_package(split_packages, contents.first)
 
             if package_to_use.nil?
               package_to_use = build_package
-              _split_packages << package_to_use
+              split_packages << package_to_use
             end
 
             package_to_use.contents << contents.shift
           end
 
-          _split_packages.drop 1 # Drop the original package to ensure only generated packages are returned
+          split_packages
```

Comment:
Why did you decide to remove `drop`? I'm not sure how it works

Reply:
I wrote about this code in PR.
I don't understand why it was implemented like this originally,

So as I understand:
1. in `reduce` method it mutates original package
2. before return from `reduce` it removes original package
3. In `split` method that package (mutated in `reduce`) is merged to generated packages `packages.push *generated_packages`

Maybe I miss something

I change it to just return generated packages with original one included

Reply:
Oh, I'm sorry. I missed it

Reply:
Just leave it as it was before. It's better to don't risk :D

Reply:
😀 haha, okay cool, I'll revert this part